### PR TITLE
Add admin guard for templates

### PR DESCRIPTION
**Pull request by yousefa00**

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
   "devDependencies": {
     "@h4iuiuc/eslint-plugin": "^1.0.12",
     "eslint": "^6.8.0",
+    "eslint-plugin-react-hooks": "^4.0.4",
```
Comment (arpanlaha) [inline api/package.json:36]: why is this here?

File: client/pages/template.jsx [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+  const implementationIndex = 2;
+
+  useEffect(() => {
+    const checkAdmin = async () => {
+      const admin = await checkIsAdmin();
+      setIsAdmin(admin);
+    };
+
+    checkAdmin();
+  }, []);
```
Comment (yousefa00) [inline client/pages/template.jsx:47]: I added this

File: client/pages/template.jsx [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+  };
+
+  return (
+    <div>
+      {
+        <div className="edit-template-div">
+          <Head title="" />
+          {!isAdmin ? (
+            <p>Only admins may view this page.</p>
+          ) : (
```
Comment (yousefa00) [inline client/pages/template.jsx:157]: and this
